test/test_testing.py

```diff
@@ -683,5 +687,99 @@ def test_calculate_2_shards_against_optimal_shards(self):
                 self.assertEqual(sorted_tests, sorted_shard_tests)
 
 
+skip_if_no_cuda = unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+
+
+class TestAsserts(TestCase):
+    def test_assert_tensors_equal_not_tensors(self):
+        a = torch.empty(())
+        b = np.empty(())
+
+        with self.assertRaises(AssertionError):
+            torch.testing.assert_tensors_equal(a, b)
+
+    @unittest.expectedFailure
+    def test_assert_tensors_equal_complex_support(self):
+        a = torch.ones(1, dtype=torch.float32)
+        b = torch.ones(1, dtype=torch.complex64)
+
+        torch.testing.assert_tensors_equal(a, b, check_dtype=False)
+
+    @unittest.expectedFailure
+    def test_assert_tensors_equal_sparse_support(self):
+        a = torch.empty(())
+        b = torch.sparse_coo_tensor(size=())
+
+        torch.testing.assert_tensors_equal(a, b)
+
+    @unittest.expectedFailure
+    def test_assert_tensors_equal_quantized_support(self):
+        a = torch.ones(1, dtype=torch.int32)
+        b = torch.ones(1, dtype=torch.qint32)
+
+        torch.testing.assert_tensors_equal(a, b)
+
+    def test_assert_tensors_equal_mismatching_shape(self):
+        a = torch.empty(())
+        b = a.clone().reshape((1,))
+
+        with self.assertRaisesRegex(AssertionError, "shape"):
+            torch.testing.assert_tensors_equal(a, b)
+
+    @skip_if_no_cuda
+    def test_assert_tensors_equal_mismatching_device(self):
+        a = torch.empty((), device=torch.device("cpu"))
+        b = a.clone().to(torch.device("cuda"))
+
+        with self.assertRaisesRegex(AssertionError, "device"):
+            torch.testing.assert_tensors_equal(a, b)
+
+    @skip_if_no_cuda
+    def test_assert_tensors_equal_mismatching_device_no_check(self):
+        a = torch.empty((), device=torch.device("cpu"))
+        b = a.clone().to(torch.device("cuda"))
+
+        torch.testing.assert_tensors_equal(a, b, check_device=False)
+
+    def test_assert_tensors_equal_mismatching_dtype(self):
+        a = torch.empty((), dtype=torch.float)
+        b = a.clone().to(torch.int)
+
+        with self.assertRaisesRegex(AssertionError, "dtype"):
+            torch.testing.assert_tensors_equal(a, b)
+
+    def test_assert_tensors_equal_mismatching_dtype_no_check(self):
+        a = torch.ones((), dtype=torch.float)
+        b = a.clone().to(torch.int)
+
+        torch.testing.assert_tensors_equal(a, b, check_dtype=False)
+
+    # def test_assert_tensors_equal_mismatching_stride(self):
+    #     a = torch.empty((1, 2))
+    #     b = a.clone().as_strided(a.shape, a.stride()[::-1])
+    #
+    #     with self.assertRaisesRegex(AssertionError, "stride"):
+    #         torch.testing.assert_tensors_equal(a, b)
+    #
+    # def test_assert_tensors_equal_mismatching_stride_no_check(self):
+    #     a = torch.ones((), dtype=torch.float)
+    #     b = torch.ones((), dtype=torch.int)
+    #
+    #     torch.testing._assert_attributes_equal(a, b, check_dtype=False)
```

Comment:
I need to yet figure out how to create two tensors with matching values but mismatching strides.